crates/bevy_ui/src/render/gradient.wgsl

```diff
@@ -0,0 +1,280 @@
+    return (flags & mask) != 0u;
+}
+
+@group(0) @binding(0) var<uniform> view: View;
+
+struct VertexOutput {
+    @location(0) uv: vec2<f32>,
+    @location(1) @interpolate(flat) size: vec2<f32>,
+    @location(2) @interpolate(flat) flags: u32,
+    @location(3) @interpolate(flat) radius: vec4<f32>,    
+    @location(4) @interpolate(flat) border: vec4<f32>,    
+
+    // Position relative to the center of the rectangle.
+    @location(5) point: vec2<f32>,
+    @location(6) @interpolate(flat) g_start: vec2<f32>,
+    @location(7) @interpolate(flat) dir: vec2<f32>,
+    @location(8) @interpolate(flat) start_color: vec4<f32>,
+    @location(9) @interpolate(flat) start_len: f32,
+    @location(10) @interpolate(flat) end_len: f32,
+    @location(11) @interpolate(flat) end_color: vec4<f32>,
+    @location(12) @interpolate(flat) hint: f32,
+    @builtin(position) position: vec4<f32>,
+};
+
+@vertex
+fn vertex(
+    @location(0) vertex_position: vec3<f32>,
+    @location(1) vertex_uv: vec2<f32>,
+    @location(2) flags: u32,
+
+    // x: top left, y: top right, z: bottom right, w: bottom left.
+    @location(3) radius: vec4<f32>,
+
+    // x: left, y: top, z: right, w: bottom.
+    @location(4) border: vec4<f32>,
+    @location(5) size: vec2<f32>,
+    @location(6) point: vec2<f32>,
+    @location(7) @interpolate(flat) g_start: vec2<f32>,
+    @location(8) @interpolate(flat) dir: vec2<f32>,
+    @location(9) @interpolate(flat) start_color: vec4<f32>,
+    @location(10) @interpolate(flat) start_len: f32,
+    @location(11) @interpolate(flat) end_len: f32,
+    @location(12) @interpolate(flat) end_color: vec4<f32>,
+    @location(13) @interpolate(flat) hint: f32
+) -> VertexOutput {
+    var out: VertexOutput;
+    out.position = view.clip_from_world * vec4(vertex_position, 1.0);
+    out.uv = vertex_uv;
+    out.size = size;
+    out.flags = flags;
+    out.radius = radius;
+    out.border = border;
+    out.point = point;
+    out.dir = dir;
+    out.start_color = start_color;
+    out.start_len = start_len;
+    out.end_len = end_len;
+    out.end_color = end_color;
+    out.g_start = g_start;
+    out.hint = hint;
+
+    return out;
+}
+
+// The returned value is the shortest distance from the given point to the boundary of the rounded 
+// box.
+// 
+// Negative values indicate that the point is inside the rounded box, positive values that the point 
+// is outside, and zero is exactly on the boundary.
+//
+// Arguments: 
+//  - `point`        -> The function will return the distance from this point to the closest point on 
+//                    the boundary.
+//  - `size`         -> The maximum width and height of the box.
+//  - `corner_radii` -> The radius of each rounded corner. Ordered counter clockwise starting 
+//                    top left:
+//                      x: top left, y: top right, z: bottom right, w: bottom left.
+fn sd_rounded_box(point: vec2<f32>, size: vec2<f32>, corner_radii: vec4<f32>) -> f32 {
```

Comment:
is this not already present in other ui code? can it be shader imported?

Reply:
Ah I thought that wgsl didn't allow importing functions, I didn't know about #define_import_path